ambry-cloud/src/test/java/com.github.ambry.cloud/CloudBlobStoreTest.java

```diff
@@ -0,0 +1,202 @@
+  private Random random = new Random();
+  private short refAccountId = 50;
+  private short refContainerId = 100;
+
+  @Before
+  public void setup() throws Exception {
+    dest = mock(CloudDestination.class);
+    partitionId = new MockPartitionId();
+    store = new CloudBlobStore(partitionId, null, dest);
+    store.start();
+  }
+
+  /** Test the CloudBlobStore put method. */
+  @Test
+  public void testStorePuts() throws Exception {
+    // Put blobs with and without expiration, in
+    // TODO: use containers flagged/not for replication
+    MockMessageWriteSet messageWriteSet = new MockMessageWriteSet();
+    int count = 10;
+    long expireTime = System.currentTimeMillis() + 10000;
+    for (int j = 0; j < count; j++) {
+      long size = Math.abs(random.nextLong()) % 10000;
+      addBlobToSet(messageWriteSet, size, Utils.Infinite_Time, refAccountId, refContainerId);
+      addBlobToSet(messageWriteSet, size, expireTime, refAccountId, refContainerId);
+    }
+    store.put(messageWriteSet);
+    verify(dest, times(count)).uploadBlob(any(BlobId.class), anyLong(), any(InputStream.class));
+  }
+
+  /** Test the CloudBlobStore delete method. */
+  @Test
+  public void testStoreDeletes() throws Exception {
+    MockMessageWriteSet messageWriteSet = new MockMessageWriteSet();
+    int count = 10;
+    for (int j = 0; j < count; j++) {
+      long size = 10;
+      addBlobToSet(messageWriteSet, size, Utils.Infinite_Time, refAccountId, refContainerId);
+    }
+    store.delete(messageWriteSet);
+    verify(dest, times(count)).deleteBlob(any(BlobId.class));
+  }
+
+  /** Test the CloudBlobStore findMissingKeys method. */
+  @Test
+  public void testFindMissingKeys() throws Exception {
+    int count = 10;
+    List<StoreKey> keys = new ArrayList<>();
+    for (int j = 0; j < count; j++) {
+      BlobId blobId = getUniqueId(refAccountId, refContainerId);
+      when(dest.doesBlobExist(eq(blobId))).thenReturn(true);
+      keys.add(blobId);
+      blobId = getUniqueId(refAccountId, refContainerId);
+      when(dest.doesBlobExist(eq(blobId))).thenReturn(false);
+      keys.add(blobId);
+    }
+    Set<StoreKey> missingKeys = store.findMissingKeys(keys);
+    verify(dest, times(count * 2)).doesBlobExist(any(BlobId.class));
+    assertEquals("Wrong number of missing keys", count, missingKeys.size());
+  }
+
+  /** Test verifying behavior when store not started. */
+  @Test
+  public void testStoreNotStarted() throws Exception {
+    // Create store and don't start it.
+    CloudBlobStore idleStore = new CloudBlobStore(partitionId, null, dest);
+    List<StoreKey> keys = Collections.singletonList(getUniqueId(refAccountId, refContainerId));
+    MockMessageWriteSet messageWriteSet = new MockMessageWriteSet();
+    addBlobToSet(messageWriteSet, 10, Utils.Infinite_Time, refAccountId, refContainerId);
+    try {
+      idleStore.put(messageWriteSet);
+      fail("Store put should have failed.");
+    } catch (StoreException storex) {
```

Comment:
Name it as `e`? Exception var in our code use `e` commonly.

Reply:
ok